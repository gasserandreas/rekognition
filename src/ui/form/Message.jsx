import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Colors } from '../../styles';

export const MESSAGE_TYPES = {
  INFO: 'info',
  ERROR: 'error',
  WARNING: 'warning',
};

const getColor = (props) => {
  switch (props.appearance) {
    case MESSAGE_TYPES.ERROR:
      return Colors.ColorsPalette.TextInvers;
    default:
      return Colors.ColorsPalette.Text;
  }
}

const getBackgroundColor = (props) => {
  switch (props.appearance) {
    case MESSAGE_TYPES.ERROR:
      return Colors.Red.Default;
    case MESSAGE_TYPES.WARNING:
      return Colors.Orange.Default;
    default:
      return 'inherit';
  }
}

const getBorderColor = (props) => {
  switch (props.appearance) {
    case MESSAGE_TYPES.ERROR:
      return Colors.Red.Default;
    case MESSAGE_TYPES.WARNING:
      return Colors.Orange.Default;
    default:
      return Colors.Neutrals.LightDark;
  }
}

/* istanbul ignore next */
const StyledMessage = styled.div`
  color: ${props => getColor(props)};
  background-color: ${props => getBackgroundColor(props)};
  border-color: ${props => getBorderColor(props)};
  border-width: ${props => props.border ? '1px' : '0'};
  border-style: solid;
  border-radius: ${props => props.rounded ? '2px' : '0'};
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  padding: 0.5rem 0.75rem;
`;

const Message = ({ children, ...props}) => (
  <StyledMessage {...props}>
    {children}
  </StyledMessage>
);

Message.propTypes = {
  appearance: PropTypes.oneOf([
    MESSAGE_TYPES.INFO,
    MESSAGE_TYPES.ERROR,
    MESSAGE_TYPES.WARNING,
  ]),
  children: PropTypes.node.isRequired,
  border: PropTypes.bool,
  rounded: PropTypes.bool,
};

Message.defaultProps = {
  appearance: MESSAGE_TYPES.INFO,
  border: false,
  rounded: true,
};

export const __testables__ = {
  getColor,
  getBorderColor,
  getBackgroundColor,
  StyledMessage,
};

export default Message;
