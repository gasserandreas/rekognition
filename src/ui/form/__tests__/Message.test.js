import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Message, { MESSAGE_TYPES } from '../Message';

it('Message should render correctly', () => {
  const wrapper = shallow(<Message><p>Hello</p></Message>);
  expect(toJson(wrapper.dive())).toMatchSnapshot();
});

it('Message should render info styles', () => {
  const wrapper = shallow(<Message appearance={MESSAGE_TYPES.INFO}><p>Hello</p></Message>);
  expect(toJson(wrapper.dive())).toMatchSnapshot();
});

it('Message should render warning styles', () => {
  const wrapper = shallow(<Message appearance={MESSAGE_TYPES.WARNING}><p>Hello</p></Message>);
  expect(toJson(wrapper.dive())).toMatchSnapshot();
});

it('Message should render error styles', () => {
  const wrapper = shallow(<Message appearance={MESSAGE_TYPES.ERROR}><p>Hello</p></Message>);
  expect(toJson(wrapper.dive())).toMatchSnapshot();
});

it('Message should render without border', () => {
  const wrapper = shallow(<Message border><p>Hello</p></Message>);
  expect(toJson(wrapper.dive())).toMatchSnapshot();
});

it('Message should render without round corners', () => {
  const wrapper = shallow(<Message rounded={false}><p>Hello</p></Message>);
  expect(toJson(wrapper.dive())).toMatchSnapshot();
});

