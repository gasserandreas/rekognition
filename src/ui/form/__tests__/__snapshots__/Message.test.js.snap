// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render correctly 1`] = `
<StyledComponent
  appearance="info"
  border={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  border-color: ",
          [Function],
          ";
  border-width: ",
          [Function],
          ";
  border-style: solid;
  border-radius: ",
          [Function],
          ";
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  padding: 0.5rem 0.75rem;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rounded={true}
>
  <p>
    Hello
  </p>
</StyledComponent>
`;

exports[`Message should render error styles 1`] = `
<StyledComponent
  appearance="error"
  border={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  border-color: ",
          [Function],
          ";
  border-width: ",
          [Function],
          ";
  border-style: solid;
  border-radius: ",
          [Function],
          ";
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  padding: 0.5rem 0.75rem;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rounded={true}
>
  <p>
    Hello
  </p>
</StyledComponent>
`;

exports[`Message should render info styles 1`] = `
<StyledComponent
  appearance="info"
  border={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  border-color: ",
          [Function],
          ";
  border-width: ",
          [Function],
          ";
  border-style: solid;
  border-radius: ",
          [Function],
          ";
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  padding: 0.5rem 0.75rem;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rounded={true}
>
  <p>
    Hello
  </p>
</StyledComponent>
`;

exports[`Message should render warning styles 1`] = `
<StyledComponent
  appearance="warning"
  border={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  border-color: ",
          [Function],
          ";
  border-width: ",
          [Function],
          ";
  border-style: solid;
  border-radius: ",
          [Function],
          ";
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  padding: 0.5rem 0.75rem;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rounded={true}
>
  <p>
    Hello
  </p>
</StyledComponent>
`;

exports[`Message should render without border 1`] = `
<StyledComponent
  appearance="info"
  border={true}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  border-color: ",
          [Function],
          ";
  border-width: ",
          [Function],
          ";
  border-style: solid;
  border-radius: ",
          [Function],
          ";
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  padding: 0.5rem 0.75rem;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rounded={true}
>
  <p>
    Hello
  </p>
</StyledComponent>
`;

exports[`Message should render without round corners 1`] = `
<StyledComponent
  appearance="info"
  border={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  border-color: ",
          [Function],
          ";
  border-width: ",
          [Function],
          ";
  border-style: solid;
  border-radius: ",
          [Function],
          ";
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  padding: 0.5rem 0.75rem;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rounded={false}
>
  <p>
    Hello
  </p>
</StyledComponent>
`;
